### PR TITLE
Require postal code if available

### PR DESCRIPTION
**Pull request by olikami**
If the postal code is available through `request.cf` it should be a requirement for the password.

### CODE REVIEW

**Comment by teh-maxh** [comment]
Including the postcode in the error message is probably too helpful.

**Comment by olikami** [comment]
This probably works for countries where the IP address data is great, but in my case it gave me the postal code of my ISP. I do want the spammers to get on to the next level 😈 .

**Comment by Greenlio** [comment]
It already has any US postal code, having their ISP postal code would be kinda weird lol?

**Comment by BitPatty** [comment]
Tried this a few times with different ISPs / devices and it never matched my location. In around 1/3 of attempts it got the coordinates of a nearby city. On cellular the data is completely off.

Even if it were accurate, IMO users shouldn't have to enter any personal/linkable metadata.

**Comment by MichaelNMaggs** [comment]
Conflicts with "Must have US Zip code" message

File: index.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ async function handleRequest(request) {
   else if(password.match('[a-z]') === null) {
     badPasswordMessage = 'Password must contain at least 1 lowercase character';
   }
+  else if (request.cf.postalCode && !password.includes(request.cf.postalCode)) {
+    badPasswordMessage = `Password must contain your postal code (${request.cf.postalCode})`
+  }
```
Comment (Saiv46) [inline index.js:27]:
```suggestion
  else if (request.cf.postalCode && !password.match(new RegExp(request.cf.postalCode.slice(0, 3) + '\d{2,4}')) {
    badPasswordMessage = `Password must contain your postal code (${request.cf.postalCode.slice(0, 3)}**)`
  }
```

As people pointed out here - postal codes does not match exact location. Yet they still match the country and city, so why not check first 3 digits instead?

**Review by Saiv46 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (Saiv46) [reply index.js:27]: In US such accuracy would give down to state, in Japan/Russia/Ukraine - down to city. In EU it still completely misses though

**Comment by troyhunt** [comment]
I've decided to drop post code out altogether in https://github.com/troyhunt/password-purgatory-api/commit/6f274b625e3cf60d52436adda2c83f78e2b622c6 as it's just too geographically specific, better we have criteria that is more globally inclusive.

**Pull request closed without merging**